### PR TITLE
fix: compiler errors cause distraction exception

### PR DESCRIPTION
**Pull request by MarkMcCulloh**
Ever since wingc stopped being a binary, invoking `compile` doesn't cause an error unless there's a panic. This means that execution isn't stopped in the CLI for regular errors but the rest of it expects the preflight js to exist, causing it to throw trying to read the file.

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Review by eladb — CHANGES_REQUESTED** [review]
Test?

File: libs/wingc/src/lib.rs [hunk 1]
```diff
@@ -78,11 +78,16 @@ pub unsafe extern "C" fn wingc_compile(ptr: u32, len: u32) {
...
-		eprintln!(
-			"Compilation failed with {} errors\n{}",
+		let result = format!(
+			"Compilation failed with {} error(s)\n{}",
 			err.len(),
 			err.iter().map(|d| format!("{}", d)).collect::<Vec<_>>().join("\n")
 		);
+
+		let leaked = result.into_bytes().leak();
+		combine_ptr_and_length(leaked.as_ptr() as u32, leaked.len() as u32)
```
Comment (yoav-steinberg) [inline libs/wingc/src/lib.rs:88]: This feels WASM only? Does this break native builds?

Comment (MarkMcCulloh) [reply libs/wingc/src/lib.rs:88]: This function is intentionally meant to only be used with WASM. There is an additional compile function that can be used by the native side with proper types.

Also with regards to breaking native builds, you may want to see [this PR](https://github.com/winglang/wing/pull/1453) if you haven't yet.

**Comment by MarkMcCulloh** [comment]
@eladb tests will be coming in https://github.com/winglang/wing/issues/926

**Comment by eladb** [comment]
> @eladb tests will be coming in #926

OK

File: libs/wingc/src/lib.rs (hunk 1 above)
Comment (Chriscbr) [inline libs/wingc/src/lib.rs:88]: If we have a `ptr_to_string` would it make sense to abstract this into a `string_to_ptr` function? I think there might be other places we do this as well

Comment (MarkMcCulloh) [reply libs/wingc/src/lib.rs]: Makes sense, it's used in two places and will be used in more over time

**Comment by MarkMcCulloh** [comment]
Recontextualized in https://github.com/winglang/wing/pull/1499

**Pull request closed without merging**